torch/distributed/_tensor/examples/torchrec_sharding_example.py

```diff
@@ -25,9 +28,77 @@ def get_device_type():
     )
 
 
-def run_torchrec_row_wise_sharding_example(rank, world_size):
-    # row-wise example:
-    #   one table is sharded by rows within the global ProcessGroup
+aten = torch.ops.aten
+supported_ops = [aten.view.default, aten._to_copy.default]
+
+
+@dataclass
+class DTensorMetadata:
+    device_mesh: DeviceMesh
+    placements: List[Placement]
+    # tensor property can be stored as TensorProperties
+
+
+# we reuse the Shard from caffe2/torch/distributed/_shard/sharded_tensor/shard.py
+@dataclass
+class TensorShard:
+    __slots__ = ["tensor", "shard_size", "shard_offset"]
+    tensor: torch.Tensor
+    shard_size: torch.Size
+    shard_offset: Tuple[int, ...]
+
+
+# this torch.Tensor subclass is a wrapper around all local shards associated
+# with a single sharded embedding table.
+class LocalShardsWrapper(torch.Tensor):
+    local_shards: List[TensorShard]
```

Comment:
I think if we use DCP's metadata, this might be simplified as:
```
local_shards: List[torch.Tensor]
storage_meta: TensorStorageMetadata
```
Then we might be able to get rid of the `TensorShard` dataclass, as ultimately what a checkpoint layer needs is the TensorStorageMetadata and its local shards